src/advanced_inv.cpp

```diff
@@ -1601,7 +1601,7 @@ void advanced_inventory::display()
                 if( by_charges ) {
                     item moving_item = g->u.reduce_charges( idx, amount_to_move );
                     assert( !moving_item.is_null() );
-                    move_cost = Pickup::cost_to_move_item( g->u, moving_item );
+                    move_cost = amount_to_move * Pickup::cost_to_move_item( g->u, moving_item );
```

Comment:
Will this cause items count by charge such as threads to have exceedingly large movement costs in the AIM?

Reply:
Yup, you are right! Thanks! I think the fix is to multiply by `moving_item.count_by_charges() ? 1 : amount_to_move` instead.

However, while looking at this I noticed that while we are using `Pickup::cost_to_move_item()`, it seems we really should use `activity_item_handling.cpp`'s `move_cost_inv()` instead. I'll have a look and try to fix all of this at the same time.

Reply:
Nah, `move_cost_inv()` is simulated cost to be used for multiple items over larger distance, it should not be used for moving one item around.